src/zc/buildout/easy_install.py

```diff
@@ -122,6 +125,46 @@ def warn(self, *args, **kw):
 
 _no_warn = _NoWarn()
 
+
+class Environment(pkg_resources.Environment):
+    """Buildout version of Environment with canonicalized names.
+
+    * pkg_resources defines the Environment class
+    * setuptools defines a PackageIndex class that inherits from Environment
+    * Buildout needs a few fixes that should be used by both.
+
+    The fixes are needed for this issue, where distributions created by
+    setuptools 69.3+ get a different name than with older versions:
+    https://github.com/buildout/buildout/issues/647
+
+    I wanted to do this with a mixin class, but somehow the changes
+    did not get picked up.  So we duplicate '__getitem__' and 'add'
+    here and in the PackageIndex class.
```

Comment:
You probably had the MixIn class already the first item in the list of base classes. – This is my usual recipe to ensure the methods of the MixIn class get picked up.

Reply:
Ah, I had expected that the method from the last class would be taken over.  I did not try it the other way around.  But the first one wins.  I have added the mixin class back.